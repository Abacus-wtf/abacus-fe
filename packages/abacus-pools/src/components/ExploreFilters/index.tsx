import React, { FunctionComponent, useEffect, useState } from "react"
import { Checkbox, Kilo, Media } from "abacus-ui"
import styled from "styled-components"
// import { usePrevious } from "@hooks/index"
// import { useSetPools } from "@state/poolData/hooks"
import { useSetPools } from "@state/poolData/hooks"
import { OrderDirection, Vault_OrderBy } from "abacus-graph"

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 40px;

  ${Media.sm`
    flex-direction: row;
    flex-wrap: wrap;
  `}
`

const Category = styled.div`
  display: flex;
  flex-direction: column;
`

const CategoryTitle = styled(Kilo)`
  margin-bottom: 16px;
  font-weight: bold;
`

const CheckboxContainer = styled.div`
  display: flex;
`

type ExploreFiltersProps = {
  page: number
}

const ExploreFilters: FunctionComponent<ExploreFiltersProps> = () => {
  const [poolSizeOrder, setPoolSizeOrder] = useState<OrderDirection | null>(
    null
  )
  const [participantsOrder, setParticipantsOrder] =
    useState<OrderDirection | null>(null)
  // const [ticketsOrder, setTicketsOrder] = useState<OrderDirection | null>(null)
  // const previousPoolSizeOrder = usePrevious(poolSizeOrder)
  // const previousPage = usePrevious(page)
  const setPools = useSetPools()

  useEffect(() => {
    if (poolSizeOrder !== null) {
      setPools(Vault_OrderBy.Size, poolSizeOrder)
    }
  }, [poolSizeOrder, setPools])

  useEffect(() => {
    if (participantsOrder !== null) {
      setPools(Vault_OrderBy.TotalParticipants, participantsOrder)
    }
  }, [participantsOrder, setPools])

  // useEffect(() => {
  //  if (ticketsOrder !== null) {
  //    setPools(Vault_OrderBy.Tickets, ticketsOrder)
  //  }
  // }, [ticketsOrder, setPools])

  return (
    <Container>
      <Category>
        <CategoryTitle>Pool Size</CategoryTitle>
        <CheckboxContainer>
          <Checkbox
            type="radio"
            name="pool_size"
            label="Low to High"
            id="pool_size_lth"
            value="Low to High"
            checked={poolSizeOrder === "asc"}
            onChange={() => setPoolSizeOrder(OrderDirection.Asc)}
          />
          <Checkbox
            type="radio"
            name="pool_size"
            label="High to Low"
            id="pool_size_htl"
            value="High to Low"
            checked={poolSizeOrder === "desc"}
            onChange={() => setPoolSizeOrder(OrderDirection.Desc)}
          />
        </CheckboxContainer>
      </Category>
      <Category>
        <CategoryTitle>Participants</CategoryTitle>
        <CheckboxContainer>
          <Checkbox
            type="radio"
            name="participants"
            label="Low to High"
            id="participants_lth"
            value="Low to High"
            checked={participantsOrder === "asc"}
            onChange={() => setParticipantsOrder(OrderDirection.Asc)}
          />
          <Checkbox
            type="radio"
            name="participants"
            label="High to Low"
            id="participants_htl"
            value="High to Low"
            checked={participantsOrder === "desc"}
            onChange={() => setParticipantsOrder(OrderDirection.Desc)}
          />
        </CheckboxContainer>
      </Category>
      {/* <Category>
        <CategoryTitle>Tickets</CategoryTitle>
        <CheckboxContainer>
          <Checkbox
            type="radio"
            name="tickets"
            label="Low to High"
            id="tickets_lth"
            value="Low to High"
            checked={ticketsOrder === "asc"}
            onChange={() => setTicketsOrder(OrderDirection.Asc)}
          />
          <Checkbox
            type="radio"
            name="tickets"
            label="High to Low"
            id="tickets_htl"
            value="High to Low"
            checked={ticketsOrder === "desc"}
            onChange={() => setTicketsOrder(OrderDirection.Desc)}
          />
        </CheckboxContainer>
  </Category> */}
    </Container>
  )
}

export default ExploreFilters
