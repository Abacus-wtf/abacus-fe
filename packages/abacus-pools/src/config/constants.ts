// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { AbstractConnector } from "@web3-react/abstract-connector"
import { InjectedConnector } from "@web3-react/injected-connector"
import Web3 from "web3"
import {
  fortmatic,
  portis,
  walletconnect,
  walletlink,
  InjectedConnectorProps,
} from "./connectors"

export declare enum ChainId {
  MAINNET = 1,
  ROPSTEN = 3,
  RINKEBY = 4,
  GÃ–RLI = 5,
  KOVAN = 42,
  MATIC = 137,
  MUMBAI = 80001,
}

export type NetworkSymbol = "ETH"

export enum NetworkSymbolEnum {
  NONE = "SOLID_NUTHIN",
  ETH = "ETH",
}

export const OPENSEA_API_KEY = process.env.GATSBY_OPENSEA_API_KEY || undefined
export const NetworkContextName = "NETWORK"
export const ETH_RPC = process.env.GATSBY_NETWORK_URL as string

export const NETWORK_CHAIN_ID = Number(process.env.GATSBY_CHAIN_ID as string)
export const IS_PRODUCTION = process.env.GATSBY_IS_PRODUCTION === "true"
export const OPENSEA_LINK = process.env.GATSBY_OPENSEA_API as string
export const BACKEND_LINK = process.env.GATSBY_APP_SERVER as string
export const GRAPHQL_ENDPOINT = process.env.GATSBY_GRAPH_API as string

export const ABC_CONTROLLER = IS_PRODUCTION
  ? ""
  : "0x40f9201c33F1974AAFd2Ad850FD071E2093b55a4"

export const ABC_TREASURY = IS_PRODUCTION
  ? ""
  : "0xB2a7d345c42bE7606f43FAA51A025fC00EC697f4"

export const ABC_FACTORY = IS_PRODUCTION
  ? ""
  : "0x0a8A03905AD54A3c8cF6Df1Fb84dbC69F43bbc51"

export const ABC_TOKEN = IS_PRODUCTION
  ? ""
  : "0x47459c3D28eb1b16B3384aC6DeCAd96b44F7C0e2"

// TODO: Rename this ABC_ALLOCATOR
export const VE_ABC_TOKEN = IS_PRODUCTION
  ? ""
  : "0x5b4d72055a8f49C12fe65A62959769f7880D47E8"

export const ABC_EPOCH = IS_PRODUCTION
  ? ""
  : "0x2d3BCfD0800034B35074fc0E995EfD65CF10C834"

export const ABC_BRIBE_FACTORY = IS_PRODUCTION
  ? ""
  : "0x8ca0cdAdeb386Ac498bE1f712e49F5f0A00C46D1"

export const ABC_CREDIT_BONDS = IS_PRODUCTION
  ? ""
  : "0x09Def36d294E8204eC2F364F2245bb3Fd7921DE4"

export const ABC_NFT_ETH = IS_PRODUCTION
  ? ""
  : "0x9F16bA20D4D7FF21150921268707E81aD2A0107c"

export const ETH_USD_ORACLE_ADDRESS =
  "0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419"

export const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000"

export const TICKET_SIZE = 1000

export interface NetworkInfo {
  rpc: any
  chainId: number
  symbol: string
  network: string
  logo: string
  blockExplorer: string
}

const NETWORK_ADDRESSES = {
  [NetworkSymbolEnum.ETH]: ETH_RPC,
}

export const web3 = (networkSymbol: NetworkSymbolEnum) =>
  new Web3(NETWORK_ADDRESSES[networkSymbol])

export const web3Eth = new Web3(ETH_RPC)

export const NetworkInfoMap: NetworkInfo[] = [
  {
    rpc: ETH_RPC,
    chainId: NETWORK_CHAIN_ID,
    symbol: IS_PRODUCTION ? "ETH" : "Rinkeby ETH",
    network: IS_PRODUCTION ? "Ethereum Mainnet" : "Rinkeby Test Network",
    logo: "eth_logo.svg",
    blockExplorer: IS_PRODUCTION
      ? "https://etherscan.io/#/"
      : "https://rinkeby.etherscan.io/#/",
  },
]

export const NetworkSymbolAndId = {
  [NETWORK_CHAIN_ID]: NetworkSymbolEnum.ETH,
}

export interface WalletInfo {
  connector?: AbstractConnector
  name: string
  iconName: string
  description: string
  href: string | null
  color: string
  primary?: true
  mobile?: true
  mobileOnly?: true
}

export const SUPPORTED_WALLETS: { [key: string]: WalletInfo } = {
  /* INJECTED: {
    connector: injected,
    name: 'Injected',
    iconName: 'arrow-right.svg',
    description: 'Injected web3 provider.',
    href: null,
    color: '#010101',
    primary: true
  }, */
  METAMASK: {
    connector: new InjectedConnector(InjectedConnectorProps),
    name: "MetaMask",
    iconName: "metamask.png",
    description: "Easy-to-use browser extension.",
    href: null,
    color: "#E8831D",
  },
  WALLET_CONNECT: {
    connector: walletconnect,
    name: "WalletConnect",
    iconName: "walletConnectIcon.svg",
    description: "Connect to Trust Wallet, Rainbow Wallet and more...",
    href: null,
    color: "#4196FC",
    mobile: true,
  },
  WALLET_LINK: {
    connector: walletlink,
    name: "Wallet Link",
    iconName: "coinbaseWalletIcon.svg",
    description: "Use Coinbase Wallet app on mobile device",
    href: null,
    color: "#315CF5",
  },
  FORTMATIC: {
    connector: fortmatic,
    name: "Fortmatic",
    iconName: "fortmaticIcon.png",
    description: "Login using Fortmatic hosted wallet",
    href: null,
    color: "#6748FF",
    mobile: true,
  },
  Portis: {
    connector: portis,
    name: "Portis",
    iconName: "portisIcon.png",
    description: "Login using Portis hosted wallet",
    href: null,
    color: "#4A6C9B",
    mobile: true,
  },
}

const DAY = 24 * 60 * 60 * 1000

export const EPOCH_LENGTH = IS_PRODUCTION ? 2 * DAY : 2 * DAY
