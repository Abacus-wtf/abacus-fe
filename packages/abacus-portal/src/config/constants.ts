import { InjectedConnector } from "@web3-react/injected-connector"
import { AbstractConnector } from "@web3-react/abstract-connector"
import Web3 from "web3"

export const COINGECKO_ETH_USD =
  "https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd"

export const NetworkContextName = "NETWORK"
export const ETH_RPC = process.env.GATSBY_NETWORK_URL as string
export const NETWORK_CHAIN_ID = Number(process.env.GATSBY_CHAIN_ID as string)
export const IS_PRODUCTION = process.env.GATSBY_IS_PRODUCTION === "true"
export const OPENSEA_LINK = process.env.GATSBY_OPENSEA_API as string
export const CURRENT_SESSIONS = IS_PRODUCTION
  ? []
  : [
      {
        address: "0x38ec00f7a966ece849f796d1d71aae2eb4c41c2d",
        tokenId: "1",
      },
    ]

export const ABC_TREASURY_ADDRESS = IS_PRODUCTION
  ? ""
  : "0x69421DCF160B85fF2586Ea2c3415b4d48AC0b729"
export const ABC_TOKEN_ADDRESS = IS_PRODUCTION
  ? ""
  : "0x5e7aD50819f507f088Aff27196e0F4B9f727fa5b"
export const ABC_AUCTION_ADDRESS = IS_PRODUCTION
  ? "0x5165a6cf99F58AA20032AbDf5aaf3F8cD6978617"
  : "0x286c881126b86519221045B9d5DaAd25E4511780"
export const ABC_PRICING_SESSION_ADDRESS = IS_PRODUCTION
  ? ""
  : "0x34F282d640AB5769bd1D868A7425ad9e48B0e10f"

export const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'

interface NetworkInfo {
  rpc: any
  chainId: number
  symbol: string
  network: string
  logo: string
}

export const web3 = new Web3(ETH_RPC)

export const NetworkInfoMap: NetworkInfo = {
  rpc: ETH_RPC,
  chainId: NETWORK_CHAIN_ID,
  symbol: "ETH",
  network: IS_PRODUCTION ? "mainnet" : "rinkeby",
  logo: "/ETH.svg",
}

export interface WalletInfo {
  connector?: AbstractConnector
  name: string
  iconName: string
  description: string
  href: string | null
  color: string
  primary?: true
  mobile?: true
  mobileOnly?: true
}

export const injected = new InjectedConnector({
  supportedChainIds: [1, 3, 4, 5, 42, 56, 137, 80001],
})

export const SUPPORTED_WALLETS: { [key: string]: WalletInfo } = {
  /*INJECTED: {
    connector: injected,
    name: 'Injected',
    iconName: 'arrow-right.svg',
    description: 'Injected web3 provider.',
    href: null,
    color: '#010101',
    primary: true
  },*/
  METAMASK: {
    connector: injected,
    name: "MetaMask",
    iconName: "metamask.png",
    description: "Easy-to-use browser extension.",
    href: null,
    color: "#E8831D",
  },
}
