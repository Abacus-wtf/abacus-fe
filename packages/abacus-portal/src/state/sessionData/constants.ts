export const PAGINATE_BY = 2
