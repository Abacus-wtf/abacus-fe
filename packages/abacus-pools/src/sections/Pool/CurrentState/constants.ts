import { IS_PRODUCTION } from "@config/constants"

export const customDurationConfig = IS_PRODUCTION
  ? {
      min: 15,
      max: 90,
    }
  : { min: 0.5, max: 7 }

export const durations = IS_PRODUCTION
  ? [
      {
        label: "15 Days",
        id: "15_day_duration",
        value: "15",
      },
      { label: "21 Days", id: "21_day_duration", value: "21" },
      { label: "30 Days", id: "30_day_duration", value: "30" },
    ]
  : [
      {
        label: "5 Minutes",
        id: "5_hour_duration",
        value: Math.round(5 / 24 / 60).toString(),
      },
      {
        label: "12 Hours",
        id: "12_hour_duration",
        value: "0.5",
      },
      { label: "2 Days", id: "2_day_duration", value: "2" },
      { label: "7 Days", id: "7_day_duration", value: "7" },
    ]
