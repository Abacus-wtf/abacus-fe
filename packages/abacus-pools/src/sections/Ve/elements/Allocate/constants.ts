import { IS_PRODUCTION } from "@config/constants"
import { VeAllocation } from "@sections/Ve/models"

export const defaultCollections: Partial<VeAllocation>[] = IS_PRODUCTION
  ? [
      {
        name: "Doodles",
        address: "0x8a90cab2b38dba80c64b7734e58ee1db38b8992e",
        imgSrc:
          "https://lh3.googleusercontent.com/7B0qai02OdHA8P_EOVK672qUliyjQdQDGNrACxs7WnTgZAkJa_wWURnIFKeOh5VTf8cfTqW3wQpozGedaC9mteKphEOtztls02RlWQ=s120",
      },
      {
        name: "CryptoPunks",
        address: "0xb47e3cd837ddf8e4c57f05d70ab865de6e193bbb",
        imgSrc:
          "https://lh3.googleusercontent.com/BdxvLseXcfl57BiuQcQYdJ64v-aI8din7WPk0Pgo3qQFhAUH-B6i-dCqqc_mCkRIzULmwzwecnohLhrcH8A9mpWIZqA7ygc52Sr81hE=s0",
      },
      {
        name: "Bored Apes",
        address: "0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d",
        imgSrc:
          "https://lh3.googleusercontent.com/Ju9CkWtV-1Okvf45wo8UctR-M9He2PjILP0oOvxE89AyiPPGtrR3gysu1Zgy0hjd2xKIgjJJtWIc0ybj4Vd7wv8t3pxDGHoJBzDB=s120",
      },
      {
        name: "Azuki",
        address: "0xed5af388653567af2f388e6224dc7c4b3241c544",
        imgSrc:
          "https://lh3.googleusercontent.com/H8jOCJuQokNqGBpkBN5wk1oZwO7LM8bNnrHCaekV2nKjnCqw6UB5oaH8XyNeBDj6bA_n1mjejzhFQUP3O1NfjFLHr3FOaeHcTOOT=s120",
      },
    ]
  : [
      {
        name: "CryptoKitties",
        address: "0x16baf0de678e52367adc69fd067e5edd1d33e3bf",
        imgSrc:
          "https://storage.googleapis.com/opensea-static/cryptokitties-logo.png",
      },
      {
        name: "Uncanny",
        address: "0x24b5a675a7684cdbb12fa5215b7b775e291ed355",
        imgSrc:
          "https://lh3.googleusercontent.com/upPM9uL5UBEUoY5hThjEgUgl8ySUC1aFL_LG7xAzQb3BFAESQ-MFbTdH36TC3vymXKYffP-9K-cFu1XcoqfSj1IKf6_k5P7nxbZy=s0",
      },
      {
        name: "Dragon Age",
        address: "0x1935899bfb630aed1fa54f2a943f0b0841724007",
        imgSrc:
          "https://lh3.googleusercontent.com/OA1RwoiShxvCf2Ib-8p_sQyCS705qLWShC0yT8ex-jbp0-snIY-e5dcI33ScOR8GDobrnN4Ofet4MEVTETh2G1chIazPMEY6iaZA_V8=s0",
      },
      {
        name: "Azuki God",
        address: "0xb74bf94049d2c01f8805b8b15db0909168cabf46",
        imgSrc:
          "https://lh3.googleusercontent.com/1jhEpK2LCYLVGst-RJHlCRpgD7agCaS8CIS8C8-x2gu2pPUgiz7yrx008ZaaEFe0lZt3UqE1R5_2EfH9n1X57ChOM5re7pFBqqs2=s0",
      },
    ]
