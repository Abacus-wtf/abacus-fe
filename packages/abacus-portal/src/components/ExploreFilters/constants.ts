import { SessionState } from "@state/sessionData/reducer"
import { BigNumber } from "ethers"
import { parseEther } from "ethers/lib/utils"

export const statuses = [
  {
    label: "Vote",
    state: SessionState.Vote,
  },
  {
    label: "Weigh",
    state: SessionState.Weigh,
  },
  {
    label: "Set Final",
    state: SessionState.SetFinalAppraisal,
  },
  {
    label: "Harvest",
    state: SessionState.Harvest,
  },
  {
    label: "Claim",
    state: SessionState.Claim,
  },
  {
    label: "Complete",
    state: SessionState.Complete,
  },
]

export type Order = "asc" | "desc" | null

export type Range = "low_range" | "mid_range" | "high_range"

export const RangeBoundaries: Record<
  Range,
  { lower_bound: BigNumber; upper_bound: BigNumber | number }
> = {
  low_range: {
    lower_bound: parseEther("0"),
    upper_bound: parseEther("0.1"),
  },
  mid_range: {
    lower_bound: parseEther("0.1"),
    upper_bound: parseEther("1"),
  },
  high_range: {
    lower_bound: parseEther("1"),
    upper_bound: Infinity,
  },
}
