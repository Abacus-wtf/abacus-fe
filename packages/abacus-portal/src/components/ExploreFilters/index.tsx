import React, { FunctionComponent, useState, useEffect } from "react"
import { Checkbox, Accordion, H2 } from "abacus-ui"
import styled from "styled-components"
import { SessionState } from "@state/sessionData/reducer"
import {
  pricingSessionWhere,
  PricingSessionFilters,
} from "@state/sessionData/queries"
import { useGetMultiSessionData } from "@state/sessionData/hooks"
import { statuses, Range, Order } from "./constants"

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
`

const HR = styled.hr`
  margin: 18px 0px;
  margin-top: 6px;
  width: 100%;
  border-color: ${({ theme }) => theme.colors.core.border};
`

let prevWhere: string = null

const ExploreFilters: FunctionComponent = () => {
  const [sessionStatuses, setSessionStatuses] = useState(new Set<number>())
  const [sortOrder, setSortOrder] = useState<Order>(null)
  const [ranges, setRanges] = useState(new Set<Range>())
  const getMultiSessionData = useGetMultiSessionData()

  const toggleSessionState = (state: SessionState) => () =>
    setSessionStatuses((statuses) => {
      if (statuses.has(state)) {
        statuses.delete(state)
      } else {
        statuses.add(state)
      }
      return new Set(statuses)
    })

  const toggleRange = (state: Range) => () =>
    setRanges((ranges) => {
      if (ranges.has(state)) {
        ranges.delete(state)
      } else {
        ranges.add(state)
      }
      return new Set(ranges)
    })

  useEffect(() => {
    const statuses = Array.from(sessionStatuses)
    const filters: PricingSessionFilters = {
      ...(statuses?.length && {
        sessionStatus: statuses,
      }),
    }
    const where = pricingSessionWhere(filters)
    if (prevWhere !== where) {
      prevWhere = where
      getMultiSessionData(where)
    }
  }, [getMultiSessionData, sessionStatuses])

  return (
    <Container>
      <H2 style={{ marginBottom: 8 }}>Explore</H2>
      <Accordion title="Status">
        {statuses.map((item) => (
          <Checkbox
            key={item.label}
            name={item.label}
            label={item.label}
            id={item.label}
            value={item.label}
            checked={sessionStatuses.has(item.state)}
            onChange={toggleSessionState(item.state)}
          />
        ))}
      </Accordion>
      <HR />
      <Accordion title="Bounty price">
        <Checkbox
          type="radio"
          name="sort_order"
          label="Low to High"
          id="Low to High"
          value="Low to High"
          checked={sortOrder === "lth"}
          onChange={() => setSortOrder("lth")}
        />
        <Checkbox
          type="radio"
          name="sort_order"
          label="High to Low"
          id="High to Low"
          value="High to Low"
          checked={sortOrder === "htl"}
          onChange={() => setSortOrder("htl")}
        />
      </Accordion>
      <HR />
      <Accordion title="Amount staked">
        <Checkbox
          name="low_range"
          label="Under 0.1Ξ"
          id="low_range"
          value="low_range"
          checked={ranges.has("low_range")}
          onChange={toggleRange("low_range")}
        />
        <Checkbox
          name="mid_range"
          label="0.1Ξ to 1Ξ"
          id="mid_range"
          value="mid_range"
          checked={ranges.has("mid_range")}
          onChange={toggleRange("mid_range")}
        />
        <Checkbox
          name="high_range"
          label="Over 1Ξ"
          id="high_range"
          value="high_range"
          checked={ranges.has("high_range")}
          onChange={toggleRange("high_range")}
        />
      </Accordion>
    </Container>
  )
}

export default ExploreFilters
