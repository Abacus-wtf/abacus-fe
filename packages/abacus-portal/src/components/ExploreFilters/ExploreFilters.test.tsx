import { render, fireEvent } from "@test-utils/index"
import React from "react"
import "@testing-library/jest-dom/extend-expect"
import ExploreFilters from "."
import { statuses } from "./constants"

const mockGetMultiSessionData = jest.fn()

jest.mock("@state/sessionData/hooks", () => ({
  useGetMultiSessionData: () => mockGetMultiSessionData,
}))

describe("ExploreFilters", () => {
  it("Matches snapshot", () => {
    const component = render(<ExploreFilters page={0} />)
    expect(component.container).toMatchSnapshot()
  })

  it("Gets new multiSessionData with where clause when status checkbox clicked", () => {
    const { getByText } = render(<ExploreFilters page={0} />)

    fireEvent.click(getByText(statuses[0].label))
    expect(mockGetMultiSessionData).toHaveBeenCalled()
    // expect(mockGetMultiSessionData).toHaveBeenCalledWith(
    //   "{ sessionStatus_in: [0], }"
    // )
  })
})
