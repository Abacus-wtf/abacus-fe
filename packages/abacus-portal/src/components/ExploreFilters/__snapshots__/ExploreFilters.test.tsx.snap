// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreFilters Matches snapshot 1`] = `
<div>
  <div
    class="sc-htJRVC dxtHNY"
  >
    <h2
      class="sc-hKwDye jQQxBx"
      style="margin-bottom: 8px;"
    >
      Explore
    </h2>
    <div
      class="sc-dkYRCH kyBTFa"
    >
      <button
        class="sc-dlVxhl sc-evcjhq gnHBnE flTZAs"
        type="button"
      >
        <svg
          class="sc-gIDmLj"
          fill="none"
          height="8"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 1 7 7 1 1"
            stroke="#1C2333"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-opacity=".72"
            stroke-width="2"
          />
        </svg>
        <div
          class="sc-furwcr sc-fHeRUh jcgyHA fTcRRu"
        >
          Status
        </div>
      </button>
      <div
        class="sc-dtDOqo bBHSVg"
      >
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="Vote"
            name="Vote"
            type="checkbox"
            value="Vote"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="Vote"
          >
            Vote
          </label>
        </div>
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="Weigh"
            name="Weigh"
            type="checkbox"
            value="Weigh"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="Weigh"
          >
            Weigh
          </label>
        </div>
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="Set Final"
            name="Set Final"
            type="checkbox"
            value="Set Final"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="Set Final"
          >
            Set Final
          </label>
        </div>
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="Harvest"
            name="Harvest"
            type="checkbox"
            value="Harvest"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="Harvest"
          >
            Harvest
          </label>
        </div>
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="Claim"
            name="Claim"
            type="checkbox"
            value="Claim"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="Claim"
          >
            Claim
          </label>
        </div>
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="Complete"
            name="Complete"
            type="checkbox"
            value="Complete"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="Complete"
          >
            Complete
          </label>
        </div>
      </div>
    </div>
    <hr
      class="sc-cbTzjv bCmYyH"
    />
    <div
      class="sc-dkYRCH kyBTFa"
    >
      <button
        class="sc-dlVxhl sc-evcjhq gnHBnE flTZAs"
        type="button"
      >
        <svg
          class="sc-gIDmLj"
          fill="none"
          height="8"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 1 7 7 1 1"
            stroke="#1C2333"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-opacity=".72"
            stroke-width="2"
          />
        </svg>
        <div
          class="sc-furwcr sc-fHeRUh jcgyHA fTcRRu"
        >
          Bounty price
        </div>
      </button>
      <div
        class="sc-dtDOqo bBHSVg"
      >
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="Low to High"
            name="sort_order"
            type="radio"
            value="Low to High"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="Low to High"
          >
            Low to High
          </label>
        </div>
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="High to Low"
            name="sort_order"
            type="radio"
            value="High to Low"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="High to Low"
          >
            High to Low
          </label>
        </div>
      </div>
    </div>
    <hr
      class="sc-cbTzjv bCmYyH"
    />
    <div
      class="sc-dkYRCH kyBTFa"
    >
      <button
        class="sc-dlVxhl sc-evcjhq gnHBnE flTZAs"
        type="button"
      >
        <svg
          class="sc-gIDmLj"
          fill="none"
          height="8"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 1 7 7 1 1"
            stroke="#1C2333"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-opacity=".72"
            stroke-width="2"
          />
        </svg>
        <div
          class="sc-furwcr sc-fHeRUh jcgyHA fTcRRu"
        >
          Amount staked
        </div>
      </button>
      <div
        class="sc-dtDOqo bBHSVg"
      >
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="low_range"
            name="low_range"
            type="checkbox"
            value="low_range"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="low_range"
          >
            Under 0.1Ξ
          </label>
        </div>
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="mid_range"
            name="mid_range"
            type="checkbox"
            value="mid_range"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="mid_range"
          >
            0.1Ξ to 1Ξ
          </label>
        </div>
        <div
          class="sc-jcFjpl erDLeq"
        >
          <input
            class="sc-iUKqMP fOkHLo"
            id="high_range"
            name="high_range"
            type="checkbox"
            value="high_range"
          />
          <label
            class="sc-caiLqq kPnhcR"
            for="high_range"
          >
            Over 1Ξ
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
